/*
 * Copyright (C) 2021 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package android.idmap2.pb;

option optimize_for = LITE_RUNTIME;

// All changes to the proto messages in this file MUST be backwards compatible. Backwards
// incompatible changes will cause previously fabricated overlays to be considered corrupt by the
// new proto message specification.
message FabricatedOverlay {
  repeated ResourcePackage packages = 1;
  string name = 2;
  string package_name = 3;
  string target_package_name = 4;
  string target_overlayable = 5;
}

message ResourcePackage {
  string name = 1;
  repeated ResourceType types = 2;
}

message ResourceType {
  string name = 1;
  repeated ResourceEntry entries = 2;
}

message ResourceEntry {
  string name = 1;
  oneof value {
    ResourceValue res_value = 2;
  }
}

message ResourceValue {
  // Corresponds with android::Res_value::dataType
  uint32 data_type = 1;
  // Corresponds with android::Res_value::data
  uint32 data_value = 2;
}